libs/gretty/src/main/groovy/org/akhikhl/gretty/AppStartTask.groovy

```diff
@@ -25,6 +27,144 @@ class AppStartTask extends StartBaseTask {
   @Delegate
   private WebAppConfig webAppConfig = new WebAppConfig()
 
+  @Input
+  @Optional
+  getContextPath() {
+    webAppConfig.contextPath
+  }
+
+  @Input
+  @Optional
+  getInitParameters() {
+    webAppConfig.initParameters
+  }
+
+  @Input
+  @Optional
+  getRealm() {
+    webAppConfig.realm
+  }
+
+  @Input
+  @Optional
+  getRealmConfigFile() {
```

Comment:
If this is a `File`, it shouldn't be `@Input`. Gradle 7.0 will actually fail the build if you do something like this.

Reply:
Thanks for the heads-up.  This property will actually be annotated by `WebAppConfigWithInputs` as well as `ServerConfigWithInputs`. As a result, the solution is to mark it `@Internal`, to agree with the declaration in your PR.